### PR TITLE
Circle functions bugfix

### PR DESCRIPTION
**Pull request by nmaslarinos**
* Fixed a bug in `intersections.intersection_circle_circle_xy ` and `_core.tangent` where a Circle's Plane was accessed instead of its center. 
* Added a `test_intersection_circle_circle_xy` in the `test_intersections`
* Fixed the `test_tangent` to work with a properly defined circle
### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involves incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by gonzalocasas** [comment]
@brgcode the failing test case is in the docstrings, but I am not sure if the docstring is wrong?

**Comment by tomvanmele** [comment]
correct formulation would be this
```python
def intersection_circle_circle_xy(circle1, circle2):
    """Calculates the intersection points of two circles in 2d lying in the XY plane.

    Parameters
    ----------
    circle1 : [plane, float] | :class:`~compas.geometry.Circle`
        Circle defined by a point, with at least XY coordinates, and a radius.
    circle2 : [plane, float] | :class:`~compas.geometry.Circle`
        Circle defined by a point, with at least XY coordinates, and a radius.

    Returns
    -------
    tuple[[float, float, float], [float, float, float]] | None
        The intersection points if there are any.
        If the circles are tangent to each other, the two intersection points are identical.
        None otherwise.

    """
    plane1, r1 = circle1
    plane2, r2 = circle2
    p1, n1 = plane1
    p2, n2 = plane2

    R = length_vector_xy(subtract_vectors_xy(p2, p1))

    if R > r1 + r2:
        return None

    if R < fabs(r1 - r2):
        return None

    if (R == 0) and (r1 == r2):
        return None

    x1, y1 = p1[:2]
    x2, y2 = p2[:2]

    R2 = R * R
    R4 = R2 * R2
    a = (r1 * r1 - r2 * r2) / (2 * R2)
    b = sqrt(2 * (r1 * r1 + r2 * r2) / R2 - (r1 * r1 - r2 * r2)**2 / R4 - 1)

    ax = (x1 + x2) / 2 + a * (x2 - x1)
    bx = b * (y2 - y1) / 2
    ix1 = ax + bx
    ix2 = ax - bx

    ay = (y1 + y2) / 2 + a * (y2 - y1)
    by = b * (x1 - x2) / 2
    iy1 = ay + by
    iy2 = ay - by

    return [ix1, iy1, 0], [ix2, iy2, 0]
```

**Comment by tomvanmele** [comment]
which gives the correct results for
```python
from compas.geometry import intersection_circle_circle_xy

c1 = ((-1, 0, 0), (0, 0, 1)), 2.0
c2 = ((+1, 0, 0), (0, 0, 1)), 2.0

x = intersection_circle_circle_xy(c1, c2)
# ([0.0, -1.7320508075688772, 0], [0.0, 1.7320508075688772, 0])

c1 = ((-1, 0, 0), (0, 0, 1)), 2.0
c2 = ((+2, -2, 0), (0, 0, 1)), 3.0

x = intersection_circle_circle_xy(c1, c2)
# ([-1.0000000000000004, -2.0, 0], [0.8461538461538461, 0.7692307692307695, 0])
```

**Comment by tomvanmele** [comment]
@nmaslarinos could you update accordingly please?

**Comment by tomvanmele** [comment]
or perhaps a little bit more readable
```python
def intersection_circle_circle_xy(circle1, circle2):
    """Calculates the intersection points of two circles in 2d lying in the XY plane.

    Parameters
    ----------
    circle1 : [plane, float] | :class:`~compas.geometry.Circle`
        Circle defined by a point, with at least XY coordinates, and a radius.
    circle2 : [plane, float] | :class:`~compas.geometry.Circle`
        Circle defined by a point, with at least XY coordinates, and a radius.

    Returns
    -------
    tuple[[float, float, float], [float, float, float]] | None
        The intersection points if there are any.
        If the circles are tangent to each other, the two intersection points are identical.
        None otherwise.

    """
    plane1, r1 = circle1
    plane2, r2 = circle2
    p1, n1 = plane1
    p2, n2 = plane2

    R = length_vector_xy(subtract_vectors_xy(p2, p1))

    if R > r1 + r2:
        return None

    if R < fabs(r1 - r2):
        return None

    if (R == 0) and (r1 == r2):
        return None

    x1, y1 = p1[:2]
    x2, y2 = p2[:2]

    cx = 0.5 * (x1 + x2)
    cy = 0.5 * (y2 + y1)

    R2 = R * R
    R4 = R2 * R2

    a = (r1 * r1 - r2 * r2) / (2 * R2)
    b = 0.5 * sqrt(2 * (r1 * r1 + r2 * r2) / R2 - (r1 * r1 - r2 * r2) ** 2 / R4 - 1)

    i1 = cx + a * (x2 - x1) + b * (y2 - y1), cy + a * (y2 - y1) + b * (x1 - x2), 0
    i2 = cx + a * (x2 - x1) - b * (y2 - y1), cy + a * (y2 - y1) - b * (x1 - x2), 0

    return i1, i2

```

**Comment by nmaslarinos** [comment]
Should the dogstring also be updated to "Circle defined by a plane..." instead of "Circle defined by a point..."?

**Comment by tomvanmele** [comment]
yes indeed

**Comment by nmaslarinos** [comment]
I have updated the definitions with all the above comments

**Comment by tomvanmele** [comment]
sorry, forgot to look at the tangent case.
the correction is similar
```python
def tangent_points_to_circle_xy(circle, point):
    """Calculates the tangent points on a circle in the XY plane.

    Parameters
    ----------
    circle : [plane, float] | :class:`~compas.geometry.Circle`
        center, radius of the circle in the xy plane.
    point : [float, float] or [float, float, float] | :class:`~compas.geometry.Point`
        XY(Z) coordinates of a point in the xy plane.

    Returns
    -------
    tuple[[float, float, 0.0], [float, float, 0.0]]
        the tangent points on the circle.

    Examples
    --------
    >>> from compas.geometry import allclose
    >>> circle = (0, 0, 0), 1.0
    >>> point = (2, 4, 0)
    >>> t1, t2 = tangent_points_to_circle_xy(circle, point)
    >>> allclose(t1, [-0.772, 0.636, 0.000], 1e-3)
    True
    >>> allclose(t2, [0.972, -0.236, 0.000], 1e-3)
    True
    """
    plane, R = circle
    center, _ = plane

    cx, cy = center[:2]
    px, py = point[:2]

    dx = px - cx
    dy = py - cy

    D = sqrt(dx**2 + dy**2)
    L2 = D**2 - R**2

    a = dx / D, dy / D
    b = -a[1], a[0]

    A = D - L2 / D
    B = sqrt(R**2 - A**2)

    t1 = cx + A * a[0] + B * b[0], cy + A * a[1] + B * b[1], 0
    t2 = cx + A * a[0] - B * b[0], cy + A * a[1] - B * b[1], 0

    return t1, t2

```

**Comment by tomvanmele** [comment]
with as test
```python
from compas.geometry import tangent_points_to_circle_xy

c = ((2, -3, 0), (0, 0, 1)), 2.0
p = (3, 0, 0)

x = tangent_points_to_circle_xy(c, p)
# ((0.930306154330093, -1.3101020514433646, 0), (3.869693845669907, -2.2898979485566358, 0))
```

**Comment by tomvanmele** [comment]
@nmaslarinos ping

**Comment by tomvanmele** [comment]
perhaps there should also be similar checks as for the intersection function

**Comment by nmaslarinos** [comment]
> with as test
> 
> ```python
> from compas.geometry import tangent_points_to_circle_xy
> 
> c = ((2, -3, 0), (0, 0, 1)), 2.0
> p = (3, 0, 0)
> 
> x = tangent_points_to_circle_xy(c, p)
> # ((0.930306154330093, -1.3101020514433646, 0), (3.869693845669907, -2.2898979485566358, 0))
> ```

When you mean as a test, do you mean a unittest file? Because I have added one file in the `tests\geometry` folder

File: src/compas/geometry/_core/tangent.py [hunk 1]
```diff
@@ -26,16 +26,17 @@ def tangent_points_to_circle_xy(circle, point):
     Examples
     --------
     >>> from compas.geometry import allclose
-    >>> circle = (0, 0, 0), 1.0
+    >>> circle = ((0, 0, 0), (1, 0, 0), (0, 1, 0)), 1.0
```
Comment (tomvanmele) [inline src/compas/geometry/_core/tangent.py]: this line is causing the issue...
should be
```python
>>> circle = ((0, 0, 0), (0, 0, 1)), 1.0
```

**Review by tomvanmele — COMMENTED** [review]
@nmaslarinos after this everything should be ok...

File: src/compas/geometry/_core/tangent.py (hunk 1 above)
Comment (nmaslarinos) [reply src/compas/geometry/_core/tangent.py]: I have updated the dogstring to the correct circle definition

**Review by tomvanmele — APPROVED** [review]

**Pull request merged**